### DISCUSSION TITLE
Modifying epub writer output to use div tags instead of section tags?

### DISCUSSION THREAD

**Discussion by riceissa** [jgm/pandoc#9409] [Q&A] [answered]
Some epub readers (e.g. SumatraPDF) don't seem to know how to jump to section headings correctly (they jump to the first page of the parent chapter, rather than the section heading itself). I noticed that if I manually modify the epub file to use div tags instead of section tags, then SumatraPDF can jump correctly to the section heading. In other words, I want to change each section (other than the h1 chapter headings) from:

```
<section id="foo" class="level2" data-number="1.2">
<h2 data-number="1.2"><span class="header-section-number">1.2</span> Foo</h2>
<p>Section content goes here.</p>
</section>
```

to:

```
<div id="foo" class="level2" data-number="1.2">
<h2 data-number="1.2"><span class="header-section-number">1.2</span> Foo</h2>
<p>Section content goes here.</p>
</div>
```

As far as I know, this can't be done using Pandoc filters because in Pandoc's native format, section headings appear on their own (like in traditional HTML output) rather than as parents containing all their content, so I can't access the "section tag" of the epub output.

**Comment by jgm** [comment, marked as answer]
Try using `-t epub2`; that should use divs if I'm not mistaken.

**Reply by riceissa** [reply]
Thank you, this worked. I had actually tried epub2 output earlier, but SumatraPDF does not like the h1 `page-break-before: always;` rule (it was inserting unnecessary blank pages) so the bookmarks were also not working on epub2, but for a different reason. With my custom CSS and epub2 output, everything is working as expected.
